skimage/segmentation/slic_superpixels.py

```diff
@@ -256,17 +270,20 @@ def slic(image, n_segments=100, compactness=10., max_iter=10, sigma=0,
 
     if spacing is None:
         spacing = np.ones(3, dtype=dtype)
-    elif isinstance(spacing, (list, tuple)):
+    else:
         spacing = np.ascontiguousarray(spacing, dtype=dtype)
 
+
+    spacing = cast(np.ndarray, spacing)
     if not isinstance(sigma, Iterable):
         sigma = np.array([sigma, sigma, sigma], dtype=dtype)
-        sigma /= spacing.astype(dtype)
+        sigma /= spacing.astype(dtype)  # type: ignore
     elif isinstance(sigma, (list, tuple)):
         sigma = np.array(sigma, dtype=dtype)
```

Comment:
Same problem as above: `sigma` is initially declared as the type `Union[float, Sequence[float]]`, but here it is assigned the result of `np.array`. The type checker continues to treat `sigma` as its declared type, and because `np.array` is function from an external untyped library and not a constructor mypy just treats it as returning `Any` which isn't  considered an error by default. So it just assumes that this is correct and sigma is still guaranteed to be of type `Union[float, Sequence[float]]`, leading to type issues further down in the code

Reply:
Awesome, thank you for clarifying this for us! 🎉